docs/dev/dev_doc_guide.md

```diff
@@ -0,0 +1,98 @@
+# How to develop developer documentation
+
+The developer documentation is needed to provide basic information about component development.
+This guide contains requirements which are applied for component developer documentation.
+
+
+## The component structure
+
+The OpenVINO component structure is fully defined in the [dedicated section of developer documentation](./index.md#openvino-component-structure),
+this guide is more concentrated on parts which are used for developer documentation, the structure you can find below:
+
+```
+openvino_component/
+    docs/               // Contains detailed component documentation
+    README.md           // Endtry point for the developer documentation
+```
+
```

Comment:
Here we need to provide information only about some parts of the component

Reply:
I would actually like to ask what do we mean by component exactly? Because according to this, the documentation is a separate OV component? Is that right?

Reply:
In general the component it is a separate part of OpenVINO toolkit. It can be user documentation, sample, transformations, frontend, plugin and etc.
Each component can contain different parts:
```
ov_component/           // Component folder
    cmake/              // (optional) CMake scripts which are related only for this component
    dev_api/            // (optional) Developer API is used in case if component wants to provide some API for internal developers
    docs/               // (optional) Contains detailed component documentation
    include/            // (optional) Public component API
    src/                // Sources of the component
    tests/              // Tests on the component
    thirdparty/         // (optional) Third-party dependencies
    CMakeLists.txt      // Main CMake script
    README.md           // (optional) Endtry point for the developer documentation
```

Developer documentation is a part of component.

if we are talking about user documentation from the developer perspective it is also a separate component which is located in the `docs/` folder. Also we have a separate cmake target for this component